DasharoModulePkg/Library/DasharoVariablesLib/DasharoVariablesLib.c

```diff
@@ -413,6 +413,90 @@ DasharoMeasureVariables (
   return Status;
 }
 
+/**
+  Enable firmware update mode (FUM) for the next boot.
+
+  @retval RETURN_SUCCESS  FUM was successfully enabled.
+**/
+EFI_STATUS
+EFIAPI
+DasharoEnableFUM (
+  VOID
+  )
+{
+  EFI_STATUS  Status;
+  BOOLEAN     Enable;
+
+  Enable = TRUE;
+  Status = gRT->SetVariable (
+      DASHARO_VAR_FIRMWARE_UPDATE_MODE,
+      &gDasharoSystemFeaturesGuid,
+      EFI_VARIABLE_BOOTSERVICE_ACCESS | EFI_VARIABLE_NON_VOLATILE,
+      sizeof (Enable),
+      &Enable
+      );
+
+  return Status;
+}
+
+/**
+  Check whether capsule updates which survive a warm system reset are permitted
+  by current configuration.
+
+  @retval TRUE   Persistent capsules can be accepted by UpdateCapsule().
+  @retval FALSE  UpdateCapsule() must fail with an error for such a capsule.
+**/
+BOOLEAN
+EFIAPI
+DasharoCapsulesCanPersistAcrossReset (
+  VOID
+  )
+{
+  EFI_STATUS  Status;
+  UINT8       MeMode;
+  UINTN       VarSize;
+
+  //
+  // Assuming there is no ME if the corresponding menu is not enabled.
+  //
+  if (!FixedPcdGetBool (PcdShowIntelMeMenu)) {

```

Comment:
Capsules do work in `config.emulation_qemu_x86_q35_uefi`, but if `config.emulation_qemu_x86_q35_uefi_all_menus` is to be used for firmware tests I thought the capsules should work there too.

Reply:
I was thinking of adding `EDK2_APU_CONFIG` and `EDK2_INTEL_ME_HAP` to allow forcing those on in `config.emulation_qemu_x86_q35_uefi_all_menus`.  QEMU config is for coreboot and this would allow to keep related changes in coreboot.  `PcdFSBClock` can be handled similarly.